File: backend/baseball/src/main/java/com/newbie/baseball/domain/game/entity/Game.java
package com.newbie.baseball.domain.game.entity;

import com.newbie.baseball.domain.team.entity.Team;
import jakarta.persistence.*;
import lombok.Getter;

@Getter
@Entity
@Table(name = "game")
public class Game {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "date", length = 10)
    private String date;

    @Column(name = "time", length = 10)
    private String time;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "home_team_id")
    private Team homeTeam;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "away_team_id")
    private Team awayTeam;

    @Column(name = "away_score")
    private Integer awayScore;

    @Column(name = "home_score")
    private Integer homeScore;

    @Column(name = "game_result", length = 10)
    private String gameResult;

    @Column(name = "stadium", length = 10)
    private String stadium;

    @Column(name = "season", length = 10)
    private String season;
}

File: backend/crawl/app/models/game.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from app.models.database import Base


# 경기 일정 테이블 정의
class Game(Base):
    __tablename__ = "game"

    id = Column(Integer, primary_key=True, index=True)
    date = Column(String(10))
    time = Column(String(10))
    away_team_id = Column(Integer, ForeignKey("team.id"))
    home_team_id = Column(Integer, ForeignKey("team.id"))
    away_score = Column(Integer)
    home_score = Column(Integer)
    game_result = Column(String(10))
    stadium = Column(String(10))
    season = Column(String(10))
    
    home_team = relationship("Team", foreign_keys=[home_team_id], back_populates="home_games")
    away_team = relationship("Team", foreign_keys=[away_team_id], back_populates="away_games")
    lineup = relationship("Lineup", back_populates="game")
    game_records = relationship("GameRecord", back_populates="game")